### PR TITLE
[hue] Added channel to enable / disable a motion sensor

### LINKED ISSUE #8389: [hue] Support activation/deactivation of motion sensor
<a alt="bounty" href="https://www.bountysource.com/issues/92746115-hue-support-activation-deactivation-of-motion-sensor"></a>

Currently the hue binding doesn't support the activation or deactivation of motion sensors.

I am aware that i could replace the inbuilt feature of the bridge with openhab rules, however i really would like that the hue system handles such small tasks by its own. So it would be nice to deactivate motion sensors if a person uses lights in a room and activate the motion sensor again at midnight or so.

Here is some information about the request to the bridge:
https://community.home-assistant.io/t/hue-motion-sensor-on-off-switch-possible/121555

This could be integrated as another channel for the motion sensor.

### PR DESCRIPTION
**Pull request by cweitkamp**
Added channel to enable / disable a motion sensor

Closes #8389 

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/sensors/PresenceHandler.java [hunk 1]
```diff
@@ -39,10 +45,38 @@
...
+                    "@text/offline.conf-error-wrong-sensor-id");
+            return;
+        }
+
+        SensorConfigUpdate configUpdate = new SensorConfigUpdate();
+        switch (channel) {
+            case CHANNEL_ENABLED:
+                configUpdate.setOn(OnOffType.ON.equals(command));
+                break;
+        }
```
Comment (lolodomo) [inline bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/sensors/PresenceHandler.java:75]: For other channels, I think you have to call the parent class (which handle few other channels).

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/sensors/PresenceHandler.java [hunk 2]
```diff
@@ -39,10 +45,38 @@
...
+
+        hueBridge.updateSensorConfig(sensor, configUpdate);
```
Comment (lolodomo) [inline bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/sensors/PresenceHandler.java:77]: It should be moved in your switch case.

**Review by lolodomo — CHANGES_REQUESTED** [review]

**Comment by lolodomo** [comment]
@cweitkamp : can you please have a look to the code of the parent class, I think we have a useless test at this line:
https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueSensorHandler.java#L193
My feeling is that the variable sensorState should have rather been set to null by default.
https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueSensorHandler.java#L183

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/sensors/PresenceHandler.java (hunk 1 above)
Comment (cweitkamp) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/sensors/PresenceHandler.java:75]: On the first glance, yes, but currently the presence sensor only has read-only channels thus no command has be handled except this one.

**Comment by cweitkamp** [comment]
@lolodomo Thanks for your comments. I saw the superfluous check and thought about an improvement too. Problem is that `StateUpdate` uses a builder pattern to add more values. If we set it to `null` initially we have to add an even more complex logic inside the switch - case elements which will reduce readability of the code.

**Review by cpmeister — APPROVED** [review]
LGTM

**Comment by cpmeister** [comment]
Looks like you need to run spotless.

**Pull request merged**